api/v1alpha4/compute_cluster.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import corev1 "k8s.io/api/core/v1"
+
+// ComputeClusterSpec defines spec for a compute cluster.
+type ComputeClusterSpec struct {
+	// SecretRef is the reference to a secret with a `kubeconfig` value
+	// providing the kubeconfig for accessing to the ComputeCluster.
+	// TODO: validation (mandatory if ComputeClusterSpec is defined; cannot be changed if the KubemarkCluster is ready)
+	KubeConfigSecretRef corev1.LocalObjectReference `json:"secretRef,omitempty"`
```

Comment:
i think it's worth mentioning where the secrete needs to be created for the controller to find it. i had some trouble figuring this out when i was testing.